main.go

```diff
@@ -117,23 +123,25 @@ func main() {
 	}
 
 	if isRunTimeDocker {
-		log.Println(logTag, "Runtime detected as docker container ...")
+		log.Println(logTag, "Runtime detected as docker or OCI container ...")
 		cmd := exec.Command("/bin/sh", "-c", "head -1 /proc/self/cgroup|cut -d/ -f3")
 		var out bytes.Buffer
 		cmd.Stdout = &out
 		err := cmd.Run()
 		id := out.String()
 		if err != nil {
-			log.Fatal(logTag, ": runtime detected as docker container: ", err)
+			log.Fatal(logTag, ": runtime detected as docker or OCI container: ", err)
 		}
 		if id == "" {
-			log.Fatal(logTag, ": runtime detected as docker container: machineid can not be empty")
+			log.Fatal(logTag, ": runtime detected as docker or OCI container: machineid can not be empty")
 		}
-		util.MachineID = strings.TrimSuffix(id, "\n")
+		h := hmac.New(sha256.New, []byte(strings.TrimSuffix(id, "\n")))
+		h.Write([]byte("reactivesearch"))
+		util.MachineID = hex.EncodeToString(h.Sum(nil))
 		util.RunTime = "Docker"
 	} else {
 		log.Println(logTag, "Runtime detected as a host machine ...")
-		id, err1 := machineid.ID()
+		id, err1 := machineid.ProtectedID("reactivesearch")
```

Comment:
@siddharthlatest Is it ok to hardcode the key? I think we can take it from env variables and use `reactivesearch` as fallback because arc code is public.

Reply:
Yes, this is how HMAC-256 works. This is a hash, we don't care about retrieving the original value.